OpenSim/ExampleComponents/Device.h

```diff
@@ -0,0 +1,94 @@
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/OpenSim.h>
+
+namespace OpenSim {
+
+//------------------------------------------------------------------------------
+// Device is a type of ModelComponent that contains all the parts comprising the
+// assistive device model (a PathActuator plus bodies and joints).
+// Devices are built by buildDevice()
+//------------------------------------------------------------------------------
+class Device : public ModelComponent {
+    OpenSim_DECLARE_CONCRETE_OBJECT(Device, ModelComponent);
+
+public:
+    // Outputs that report quantities in which we are interested.
+    // The total length of the device.
+    OpenSim_DECLARE_OUTPUT(length, double, getLength, SimTK::Stage::Position);
+
+    // The lengthening speed of the device.
+    OpenSim_DECLARE_OUTPUT(speed, double, getSpeed, SimTK::Stage::Velocity);
+
+    // The force transmitted by the device.
+    OpenSim_DECLARE_OUTPUT(tension, double, getTension, SimTK::Stage::Dynamics);
+
+    // The power produced(+) or dissipated(-) by the device.
+    OpenSim_DECLARE_OUTPUT(power, double, getPower, SimTK::Stage::Dynamics);
+
+    // The height of the model to which the device is attached.
+    OpenSim_DECLARE_OUTPUT(height, double, getHeight, SimTK::Stage::Position);
+
+    // The center of mass height of the model to which the device is attached.
+    OpenSim_DECLARE_OUTPUT(com_height, double, getCenterOfMassHeight,
+                           SimTK::Stage::Position);
+
+    // Member functions that access quantities in which we are interested. These
+    // methods are used by the outputs declared above.
+    double getLength(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").getLength(s);
+    }
+    double getSpeed(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").getLengtheningSpeed(s);
+    }
+    double getTension(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").computeActuation(s);
+    }
+    double getPower(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").getPower(s);
+    }
+    double getHeight(const SimTK::State& s) const {
+        static const std::string hopperHeightCoord = "/Dennis/slider/yCoord";
```

Comment:
My expectation was that this class would still be specific to the hopper example; specifically the MATLAB hopper example. Perhaps it would be sufficient to document this in the class description (e.g., "This class is for use with the MATLAB Hopper_Device example, and is not general enough to be used elsewhere"), and then leave in parts like this?